### PR TITLE
Fix possible silent failure to connect to workflow

### PR DESCRIPTION
**Pull request by MetRonnie**
If you played a workflow with the same workflow ID as a previously existing one, but a new UUID, the UI Server was not connecting to it.

Due to https://github.com/cylc/cylc-flow/issues/5615 this was affecting restarting any workflow.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes
- [ ] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] No docs needed
- [ ] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by MetRonnie** [comment]
Not sure about testing this one

File: CHANGES.md [hunk 1]
```diff
@@ -12,10 +12,14 @@ updated. Only the first match gets replaced, so it's fine to leave the old
 ones in. -->
 
 -------------------------------------------------------------------------------
-## __cylc-uiserver-1.2.3 (<span actions:bind='release-date'>UPCOMING</span>)__
+## __cylc-uiserver-1.3.0 (<span actions:bind='release-date'>Upcoming</span>)__
 
-[#456](https://github.com/cylc/cylc-uiserver/pull/456)- (packaging) pinned
-upper version of jupyter_server to prevent `pip` from installing version 2.
```
Comment (MetRonnie) [inline CHANGES.md:18]: This is going to be wiped out by #450

File: cylc/uiserver/workflows_mgr.py [hunk 2]
```diff
@@ -305,37 +305,40 @@ def run(*coros):
...
-                    self._unregister(wid),
-                )
-
-            elif before is None and after == 'active':
-                # workflow has been created and started
-                run(
-                    self._register(wid, flow, is_active=True),
-                    self._connect(wid, flow),
-                )
+            if before == 'active':
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/uiserver/workflows_mgr.py:308]: Whereas I feel that I'm having to draw a state table to understand. It's a style argument though.

Surely whether it takes the CPU longer depends on whether you order the options sensibly? If you do

```
if likely and common:
  ...
elif likely and rare:
  ...
elif unlikely and common:
  ...
elif unlikely and rare:
  ...
```
Will be better than
```
if unlikely:
  if rare:
    ...
  if common:
    ...
if likely:
  if rare:
    ...
  if common:
    ...
```
(there are a ton of assumptions about execution order being preserved when the code is actually run through the interpreter tho)

[the remark this answers is not in the document]
Comment (MetRonnie) [reply cylc/uiserver/workflows_mgr.py:308]: (Anyway I don't think we have to worry about the execution time - it's possible the comparison operation result is cached in the CPU for all I know)

[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/uiserver/workflows_mgr.py:308]: I'm not holding the review for this minor style point.

[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/uiserver/workflows_mgr.py:308]: (I think the reason I wrote it out flat was that I was listing every legal state change)

**Review by oliver-sanders — APPROVED** [review]

**Review by wxtim — APPROVED** [review]

**Pull request merged**